v2/pkg/subscraping/sources/netlas/netlas.go

```diff
@@ -0,0 +1,196 @@
+// Package netlas logic
+package netlas
+
+import (
+	"context"
+
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strconv"
+	"time"
+
+	"github.com/projectdiscovery/subfinder/v2/pkg/subscraping"
+)
+
+type Item struct {
+	Data struct {
+		A           []string `json:"a,omitempty"`
+		Txt         []string `json:"txt,omitempty"`
+		LastUpdated string   `json:"last_updated,omitempty"`
+		Timestamp   string   `json:"@timestamp,omitempty"`
+		Ns          []string `json:"ns,omitempty"`
+		Level       int      `json:"level,omitempty"`
+		Zone        string   `json:"zone,omitempty"`
+		Domain      string   `json:"domain,omitempty"`
+		Cname       []string `json:"cname,omitempty"`
+		Mx          []string `json:"mx,omitempty"`
+	} `json:"data"`
+}
+
+type Response struct {
+	Items []Item `json:"items"`
+	Took  int    `json:"took"`
+}
+
+type DomainsCountResponse struct {
+	Count int `json:"count"`
+}
+
+// Source is the passive scraping agent
+type Source struct {
+	apiKeys   []string
+	timeTaken time.Duration
+	errors    int
+	results   int
+	skipped   bool
+}
+
+// func get_domains(offset string, domain string) int {
+
+// 	return resp.StatusCode
+// }
+
+func (s *Source) Run(ctx context.Context, domain string, session *subscraping.Session) <-chan subscraping.Result {
+	results := make(chan subscraping.Result)
+	s.errors = 0
+	s.results = 0
+
+	go func() {
+		defer func(startTime time.Time) {
+			s.timeTaken = time.Since(startTime)
+			close(results)
+		}(time.Now())
+
+		// To get count of domains
+		endpoint := "https://app.netlas.io/api/domains_count/"
+		paramss := url.Values{}
+		paramss.Set("q", "domain:(domain:*."+domain+" "+"AND NOT domain:"+domain+")")
```

Comment:
Alternatively, you can improve its readability by using `fmt.Sprintf`:
```
query := fmt.Sprintf("domain:(domain:*.%s AND NOT domain:%s)", domain, domain)
paramss.Set("q", query)

```

Reply:
I changed the code as per your suggestion.